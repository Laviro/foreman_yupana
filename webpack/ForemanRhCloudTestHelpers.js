export const inventoryStateWrapper = innerState => ({
  ForemanRhCloud: {
    inventoryUpload: { ...innerState },
  },
});
