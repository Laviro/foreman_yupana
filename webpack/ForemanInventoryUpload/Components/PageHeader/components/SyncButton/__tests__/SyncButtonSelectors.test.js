import { testSelectorsSnapshotWithFixtures } from 'react-redux-test-utils';
import { inventoryStateWrapper } from '../../../../../../ForemanRhCloudTestHelpers';
import {
  status,
  error,
  syncHosts,
  disconnectHosts,
} from './SyncButtonFixtures';
import {
  selectInventorySync,
  selectStatus,
  selectError,
  selectSyncHosts,
  selectDisconnectHosts,
} from '../SyncButtonSelectors';

const state = inventoryStateWrapper({
  inventorySync: {
    status,
    error,
    syncHosts,
    disconnectHosts,
  },
});

const fixtures = {
  'should return InventorySync': () => selectInventorySync(state),
  'should return InventorySync status': () => selectStatus(state),
  'should return InventorySync error': () => selectError(state),
  'should return InventorySync SyncHosts': () => selectSyncHosts(state),
  'should return InventorySync disconnectHosts': () =>
    selectDisconnectHosts(state),
};

describe('SyncButton selectors', () =>
  testSelectorsSnapshotWithFixtures(fixtures));
