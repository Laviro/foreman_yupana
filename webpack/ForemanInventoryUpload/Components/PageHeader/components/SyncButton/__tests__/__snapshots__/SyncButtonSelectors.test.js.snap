// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SyncButton selectors should return InventorySync 1`] = `
Object {
  "disconnectHosts": 0,
  "error": "some-error",
  "status": "RESOLVED",
  "syncHosts": 1,
}
`;

exports[`SyncButton selectors should return InventorySync SyncHosts 1`] = `1`;

exports[`SyncButton selectors should return InventorySync disconnectHosts 1`] = `0`;

exports[`SyncButton selectors should return InventorySync error 1`] = `"some-error"`;

exports[`SyncButton selectors should return InventorySync status 1`] = `"RESOLVED"`;
